autonat.go

```diff
@@ -46,6 +39,15 @@ type AmbientAutoNAT struct {
 
 	emitReachabilityChanged event.Emitter
 	subscriber              event.Subscription
+
+	addrsSet *addrset.DialAddrsSet
+	// returns addresses that have been confirmed to be public by the AutoNAT server
+	publicAddrsManager *dialedAddrsManager
+	// returns addresses that have been confirmed to be private by the AutoNAT server
+	privateAddrsManager *dialedAddrsManager
+
+	// current NAT status and known dialable addresses.
+	status atomic.Value
```

Comment:
I haven't dug into this, but it looks like there's going to be quite a bit of duplicate state here:

* addrSet will mostly duplicate the information in the addr managers, right?
* status just reflects the sum of the information in these sets.

Does this split exist for performance?